vispy/visuals/axis.py

```diff
@@ -171,7 +171,9 @@ def _prepare_draw(self, view):
             # TODO: make sure we only call get_transform if the transform for
             # the line is updated
             tr = self._line.get_transform(map_from='visual', map_to='canvas')
-            x1, y1, x2, y2 = tr.map(self.pos)[:, :2].ravel()
+            trpos = tr.map(self.pos)
+            trpos /= trpos[:, 3:]
```

Comment:
What is this line actually doing?

Reply:
This is the fix that is required for the resulting display coordinates to be correct in the presence of non-zero FOV in the turntable camera. I wrote this a few years ago so I don't remember how exactly it works 🤔

Reply:
I'm trying to figure out what the `3:` is for. If `trpos` is (N, 4)` then I suppose this is dividing `x, y, z` coordinates by a `w` coordinate? I feel like I'm missing something obvious.

Reply:
Yes, we're basically renormalizing the coordinates based on this last fourth coordinate - I could have written it ``trpos[:, 3]`` instead of ``3:``. I can't remember the details of this but when there are projection effects the last ``w`` coordinate ends up being != 1.

Reply:
@djhoese - ah I found this old issue that discusses it! https://github.com/vispy/vispy/issues/1198#issuecomment-212790199

Reply:
Ah, so the answer is math. Got it. (ok, fine, homogeneous coordinates)